### PR TITLE
chore: get fee asset refactor

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

LLM generated description:

This PR refactors the selectFeeAssetByChainId and selectFeeAssetById selector functions to extract their business logic into separate pure functions. The new pure functions, getFeeAssetByChainId and getFeeAssetByAssetId, have been added to the utils.ts file.

The changes made in this PR include:

Create getFeeAssetByChainId and getFeeAssetByAssetId pure functions in the utils.ts file, with the specified function signatures and business logic from the original selector functions.

Update the selectFeeAssetByChainId and selectFeeAssetById selector functions in the selectors.ts file to use the new pure functions.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

n/a - this has been pissing me off for a long time

## Risk

minimal - biz logic extraction only

### Engineering

ensure fn sig changes are sane

### Operations

not testable by ops

## Screenshots (if applicable)

n/a

### CODE REVIEW

File: src/state/slices/assetsSlice/utils.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        throw new Error(`Chain namespace ${chainNamespace} on ${chainReference} not supported.`)
+    }
+  })()
+}
+
+type GetFeeAssetByChainId = (assetsById: AssetsById, chainId: ChainId) => Asset | undefined
+type GetFeeAssetByAssetId = (assetsById: AssetsById, assetId: AssetId) => Asset | undefined
+
+export const getFeeAssetByChainId: GetFeeAssetByChainId = (assetsById, chainId) => {
+  if (!chainId) return undefined
```
Comment (0xApotheosis) [inline src/state/slices/assetsSlice/utils.ts:73]: If this can be `undefined`, is our function signature wrong?

**Review by 0xApotheosis — DISMISSED** [review]
Looks sane, tested with no regressions. One question.

File: src/state/slices/assetsSlice/utils.ts (hunk 1 above)
Comment (0xdef1cafe) [reply src/state/slices/assetsSlice/utils.ts:73]: indeed - will update

Comment (0xdef1cafe) [reply src/state/slices/assetsSlice/utils.ts:80]: https://github.com/shapeshift/web/pull/4198/commits/87761fb6d280d1edc409aeb7e42d65026831c6c6

**Review by gomesalexandre — APPROVED** [review]
Untested but conceptually LLMLGTM, ty for the hyuge quality of life improvement

**Pull request merged**